src/center-common/app/models/owner_change_request.rb

```diff
@@ -0,0 +1,145 @@
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+# Request for adding client to a security server.
+# Fields: security_server, sec_serv_user, origin
+# Usage:
+# OwnerChangeRequest.new(
+#     :security_server => serverId,
+#     :sec_serv_user => clientId,
+#     :origin => origin).register()
+class OwnerChangeRequest < RequestWithProcessing
+  before_create do |rec|
+    Request.set_server_owner_name(rec)
+    Request.set_server_user_name(rec)
+  end
+
+  def self.find_by_server_and_client(server_id, client_id,
+      processing_status = nil)
+    Request.find_by_server_and_client(OwnerChangeRequest, server_id, client_id,
+        processing_status)
+  end
+
+  # Revokes owner change request with specific database id. Raises error if not
+  # found.
+  def self.revoke(request_id)
+    owner_change_request = OwnerChangeRequest.find(request_id)
+
+    unless owner_change_request
+      raise "No owner change request with id '#{request_id}'"
+    end
+
+    if !reg_request.can_revoke?
+      raise "Cannot revoke owner change request "\
+        "in state '#{owner_change_request.request_processing.status}' and with origin "\
+        "'#{owner_change_request.origin}'"
+    end
+
+  end
+
+  def find_processing
+    OwnerChangeRequestProcessing.find_by_server_and_client(
+        security_server, sec_serv_user)
+  end
+
+  def new_processing
+    OwnerChangeRequestProcessing.new
+  end
+
+  def verify_request()
+    if from_center?
+      require_client(security_server.owner_id)
+      require_security_server(security_server)
+    end
+
+    verify_against_submitted_requests()
+    validate_request()
+  end
+
+  def verify_against_submitted_requests()
+    processings = OwnerChangeRequest.find_by_server_and_client(
+        security_server, sec_serv_user,
+        RequestProcessing::SUBMITTED_FOR_APPROVAL)
+    if processings != nil and not processings.empty?
+      previous_request = processings.first
+
+      raise InvalidOwnerChangeRequestException.new(
+          I18n.t("requests.duplicate_owner_change_requests",
+            :user => sec_serv_user,
+            :security_server => security_server,
+            :received => previous_request.created_at,
+            :id => previous_request.id))
+    end
+  end
+
+  def validate_request
```

Comment:
It probably should be verified that the new owner is a registered client of the security server.

Reply:
Fixed in the commit 86f9e83.